addon-test-support/ember-mocha/index.js

```diff
@@ -6,8 +6,28 @@ import { loadTests } from './test-loader';
 import setupTest from 'ember-mocha/setup-test';
 import setupRenderingTest from 'ember-mocha/setup-rendering-test';
 import setupApplicationTest from 'ember-mocha/setup-application-test';
-import { it, afterEach } from 'mocha';
+import { beforeEach, afterEach } from 'mocha';
 import { setResolver, resetOnerror } from '@ember/test-helpers';
+import Ember from 'ember';
+
+/**
+  Ensures that `Ember.testing` is set to `true` before each test begins
+  (including `before` / `beforeEach`), and reset to `false` after each test is
+  completed. This is done via `beforeEach` and `afterEach`.
+ */
+export function setupEmberTesting() {
+  beforeEach(function() {
+    Ember.testing = true;
+  });
+
+  afterEach(function() {
+    Ember.testing = false;
+  });
+}
```

Comment:
We should not need to add this code. It should be handled by `@ember/test-helpers`

https://github.com/emberjs/ember-test-helpers/blob/d9bd493b204b6856d536f6a5d253323c3dd95686/addon-test-support/%40ember/test-helpers/setup-context.ts#L173

https://github.com/emberjs/ember-test-helpers/blob/d9bd493b204b6856d536f6a5d253323c3dd95686/addon-test-support/%40ember/test-helpers/setup-context.ts#L143

---

Also, `before` should **not** have `Ember.testing === true`, only while running an actual test (e.g. `beforeEach`).